### PR TITLE
feat(player-manager): add mini player OSD

### PR DESCRIPTION
**Pull request by MrTimscampi**
Adds a snazzy OSD to the mini player, allowing you to stop playback, move to the big player, pause/unpause and skip tracks.

Also fixes a bunch of bugs in the playback manager in order to get this working.

image

### CODE REVIEW

File: components/VideoPlayer.vue [hunk 1]
```diff
@@ -90,6 +91,19 @@ export default Vue.extend({
...
+            case 'playbackManager/PAUSE_PLAYBACK':
+              (this.$refs.videoPlayer as HTMLVideoElement).pause();
+              break;
+            case 'playbackManager/UNPAUSE_PLAYBACK':
+              (this.$refs.videoPlayer as HTMLVideoElement).play();
+              break;
+            case 'playbackManager/RESET_CURRENT_TIME':
+              (this.$refs.videoPlayer as HTMLVideoElement).currentTime = 0;
+          }
+        });
```
Comment (Camc314) [inline components/VideoPlayer.vue:106]: when any of the mutations are called, it is erroring out:
`Cannot read property 'pause' of undefined`

**Review by Camc314 — COMMENTED** [review]
I can't seem to play/pause/makefull screen/exit the video

File: components/VideoPlayer.vue (hunk 1 above)
Comment (MrTimscampi) [reply components/VideoPlayer.vue]: I can't seem to reproduce it on either Chromium or Firefox. I added null checks, so we don't trigger it accidentally.

Pause works here, but the close/full button had some issues due to a div being in front of them. So I added some pointer-event stuff in CSS to fix it. That fixes it for me.

Comment (Camc314) [reply components/VideoPlayer.vue]: Strange, I've retested and it all seems to be working now.

**Review by Camc314 — APPROVED** [review]

**Pull request merged**